internal/cmd/providers.go

```diff
@@ -36,100 +37,166 @@ func providersCmd() *cobra.Command {
 
 func createProviderCmd() *cobra.Command {
 	var (
-		hostname       string
-		clusterAddress string
+		fromContext   bool
+		clusterCA     string
+		clusterAddr   string
+		SAToken       string
+		namespace     string
+		storageClass  string
+		sshEnabled    bool
+		environmentSA string
+		orgAllowlist  []string
 	)
 	cmd := &cobra.Command{
-		Use:   "create [name] --hostname=[hostname] --cluster-address=[clusterAddress]",
+		Use:   "create [name]",
 		Args:  xcobra.ExactArgs(1),
 		Short: "create a new workspace provider.",
 		Long:  "Create a new Coder workspace provider.",
-		Example: `# create a new workspace provider in a pending state
+		Example: `# create a new workspace provider
 
-coder providers create my-provider --hostname=https://provider.example.com --cluster-address=https://255.255.255.255`,
+coder providers create my-provider`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			ctx := cmd.Context()
+			var (
+				ctx  = cmd.Context()
+				name = args[0]
+				err  error
+				req  = &coder.WorkspaceProviderKubernetesCreateRequest{
+					Name:             name,
+					ClusterCA:        clusterCA,
+					ClusterAddress:   clusterAddr,
+					SAToken:          SAToken,
+					DefaultNamespace: namespace,
+					StorageClass:     storageClass,
+					SSHEnabled:       sshEnabled,
+					EnvironmentSA:    environmentSA,
+					OrgAllowlist:     orgAllowlist,
+				}
+			)
+
+			if fromContext {
+				context, err := xkube.CurrentKubeContext()
+				if err != nil {
+					return xerrors.Errorf("reading kube config from context: %w", err)
+				}
+				if req.DefaultNamespace == "" {
+					req.DefaultNamespace = context.Namespace
+				}
+				if req.ClusterAddress == "" {
+					req.ClusterAddress = context.ClusterAddress
+				}
+				sa, err := xkube.CoderServiceAccountFromContext(ctx)
+				if err != nil {
+					return xerrors.Errorf("reading coder kubernetes service account: %w", err)
+				}
+				if req.SAToken == "" {
+					req.SAToken = sa.SAToken
+				}
+				if req.ClusterCA == "" {
+					req.ClusterCA = sa.ClusterCA
+				}
+			}
 
 			client, err := newClient(ctx, true)
 			if err != nil {
 				return err
 			}
 
-			version, err := client.APIVersion(ctx)
+			_, err = client.CreateWorkspaceProvider(ctx, *req)
 			if err != nil {
-				return xerrors.Errorf("get application version: %w", err)
+				return err
 			}
 
-			cemanagerURL := client.BaseURL()
-			ingressHost, err := url.Parse(hostname)
-			if err != nil {
-				return xerrors.Errorf("parse hostname: %w", err)
-			}
+			return nil
+		},
+	}
 
-			if cemanagerURL.Scheme != ingressHost.Scheme {
-				return xerrors.Errorf("Coder access url and hostname must have matching protocols: coder access url: %s, workspace provider hostname: %s", cemanagerURL.String(), ingressHost.String())
-			}
+	cmd.Flags().BoolVar(&fromContext, "from-context", false, "use current kube context to retrieve cluster-cert, cluster-address, sa-token, and namespace if not set explicitly")
```

Comment:
can we match the config name used by Kubernetes (kubectl or helm) when possible

```
$ kubectl options
The following options can be passed to any command:

      --cluster='': The name of the kubeconfig cluster to use
      --context='': The name of the kubeconfig context to use
      --kubeconfig='': Path to the kubeconfig file to use for CLI requests.
  -n, --namespace='': If present, the namespace scope for this CLI request
used to contact the server is used
      --token='': Bearer token for authentication to the API server
```

or Helm, which uses `--kube-context`

```
$ helm install --help
      --kube-apiserver string       the address and the port for the Kubernetes API server
      --kube-as-group stringArray   group to impersonate for the operation, this flag can be repeated to specify multiple groups.
      --kube-as-user string         username to impersonate for the operation
      --kube-ca-file string         the certificate authority file for the Kubernetes API server connection
      --kube-context string         name of the kubeconfig context to use
      --kube-token string           bearer token used for authentication
      --kubeconfig string           path to the kubeconfig file
  -n, --namespace string            namespace scope for this request
```